assets/app/view/game/buy_trains.rb

```diff
@@ -129,10 +129,21 @@ def render
       end
 
       def from_depot(depot_trains)
+        trains_from_depot = buyable_trains_in_depot(depot_trains)
+        @corporation.abilities(:train_discount) do |ability|
```

Comment:
why do we need to do this separately, why can't you just call buyable_trains_in_depot one time?

Reply:
This is to get different prices for the same train.
See included picture
![Screenshot from 2020-07-23 13-41-46](https://user-images.githubusercontent.com/2631151/88282797-7f659a00-ccea-11ea-9eed-75cef841affc.png)

Reply:
In latest versions the only thing that changes in the view is the price shown. So this picture is not relevant any longer.